lib/rules/no-arrow-function-lifecycle.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @fileoverview It is not necessary to use arrow function for lifecycle methods
+ * @author Tan Nguyen
+ */
+'use strict';
+
+const Components = require('../util/Components');
+const astUtil = require('../util/ast');
+const docsUrl = require('../util/docsUrl');
+
+const lifecycleMethods = [
+  'getDefaultProps',
+  'getInitialState',
+  'getChildContext',
+  'getDerivedStateFromProps',
+  'componentWillMount',
+  'UNSAFE_componentWillMount',
+  'componentDidMount',
+  'componentWillReceiveProps',
+  'UNSAFE_componentWillReceiveProps',
+  'shouldComponentUpdate',
+  'componentWillUpdate',
+  'UNSAFE_componentWillUpdate',
+  'getSnapshotBeforeUpdate',
+  'componentDidUpdate',
+  'componentDidCatch',
+  'componentWillUnmount',
+  'render'
+];
```

Comment:
Most of this list is already defined elsewhere, should we make this list shareable so its easier to maintain?

https://github.com/yannickcr/eslint-plugin-react/blob/7dce90a4f570847b33eaefddb8d9bb7d6eca9dc5/lib/rules/no-typos.js#L14-L29

https://github.com/yannickcr/eslint-plugin-react/blob/6b239a7c71eb2d7060edd5f48f74f138ff89736e/lib/rules/no-unused-prop-types.js#L23-L24